packages/model-serving/src/concepts/ModelDeploymentsContext.tsx

```diff
@@ -151,29 +155,26 @@ export const ModelDeploymentsProvider: React.FC<ModelDeploymentsProviderProps> =
 
   return (
     <ModelDeploymentsContext.Provider value={contextValue}>
-      {
-        // the only way to dynamically call hooks (useWatchDeployments) is to render them in dynamic components
-        deploymentWatchers.map((watcher) => {
-          const platformId = watcher.properties.platform;
-
-          if (!deploymentWatchersLoaded) {
-            return null;
-          }
-
-          return (
-            <PlatformDeploymentWatcher
-              key={platformId}
-              platformId={platformId}
-              watcher={watcher}
-              projects={projects}
-              labelSelectors={labelSelectors}
-              onStateChange={updatePlatformDeployments}
-              unloadPlatformDeployments={unloadPlatformDeployments}
-              filterFn={filterFn}
-            />
-          );
-        })
-      }
+      {deploymentWatchers.map((watcher) => {
+        const platformId = watcher.properties.platform;
+
+        if (!deploymentWatchersLoaded) {
+          return null;
+        }
+
+        return (
+          <PlatformDeploymentWatcher
+            key={platformId}
+            platformId={platformId}
+            watcher={watcher}
+            projects={projects ?? []}
```

Comment:
The infinite rerender might be caused by the ` ?? []`, the `[]` is a new object reference on every rerender. 


```suggestion
            projects={projects}
```